env/harbor/runner.py

```
@@ -0,0 +1,717 @@
+        driver = self._docker_output("info", "--format", "{{.Driver}}")
+        if driver != "fuse-overlayfs":
+            raise RuntimeError(
+                f"nested dockerd uses {driver!r}, expected 'fuse-overlayfs'"
+            )
+        return driver
+
+    def materialize_bundle(self, spec: RunSpec) -> RunSpec:
+        if spec.bundle_package_dir is None:
+            return spec
+        output_dir = RUNTIME_DIR / f"bundle-{_safe_name(spec.session_id)}"
+        subprocess.run(
```

Ruff (0.16.1)

[error] `subprocess` call: check for execution of untrusted input

(S603)